sound/soc/soc-core.c

```diff
@@ -976,7 +976,6 @@ static void soc_set_name_prefix(struct snd_soc_card *card,
 static void soc_cleanup_component(struct snd_soc_component *component)
 {
 	snd_soc_component_set_jack(component, NULL, NULL);
-	list_del(&component->card_list);
```

Comment:
@bardliao did you actually got a crash for this?

Reply:
@dbaluta Yes, I hit the issue when codec driver return -EPROBE_DEFER from its probe function.

Reply:
@bardliao You could use list_init early on before adding the component to the list. In that case list_del will work just fine and not cause crashes of any kind (doing list_del on an empty list_head is a no-op).

Reply:
@paulstelian97 Yes, ``INIT_LIST_HEAD(&component->card_list);`` is called by snd_soc_component_initialize() which is before soc_probe_component(). And yes, it is fine if we call list_del() once, but it will crash if we call it twice. When codec probe return -EPROBE_DEFER, kernel will try to re probe it again, and it will crash at the second probe failed.